src/lib/tools.ts

```diff
@@ -0,0 +1,50 @@
+import { SerpAPI } from 'langchain/tools';
+import { Calculator } from 'langchain/tools/calculator';
+import { DynamicTool, DynamicStructuredTool } from 'langchain/tools';
+import { QdrantClient } from '@qdrant/js-client-rest';
+import { OpenAIEmbeddings } from 'langchain/embeddings/openai';
+import { z } from 'zod';
+
+function getTools() {
+	if (import.meta.env.VITE_SERPAPI_API_KEY == undefined) {
```

Comment:
This is good to do this, I actually think we should do this with all of our API keys. We might want to have some code which does this in server.ts maybe as a separate PR.